### ISSUE TITLE
When should training be stopped?

### ISSUE THREAD

**Issue by YY-RR-ZZ** [TencentARC/T2I-Adapter#133]
Hello,Thank you very much for your open-source work. 
I have a question regarding the visualization of the loss generated while running train_sketch.py. Below is the visualization of the validation loss. The training reached 290,000 steps and epoch 7, but the loss does not show signs of convergence. I am unsure when the training should be considered complete.
Could you please provide some guidance on this?
![loss](https://github.com/user-attachments/assets/60e4023e-d04f-40eb-ba14-42a5d0455d9b)

**Comment by Yonghao-Yu** [comment]
Where did you download the training data from?
data/LAION_6plus
data/WebDataset

**Comment by cuhkfz** [comment]
@YY-RR-ZZ 
Hello, I encountered the same problem. Have you finished the training successfully?

**Comment by cuhkfz** [comment]
@YY-RR-ZZ 
Hello, I encountered the same problem. Have you finished the training successfully?

**Comment by Yonghao-Yu** [comment]
> @YY-RR-ZZ Hello, I encountered the same problem. Have you finished the training successfully?

hi，how you get the train data？
data/LAION_6plus
data/WebDataset

**Comment by cuhkfz** [comment]
@Yonghao-Yu Hello yonghao, I use COCO dataset, the original T2I Adapter with sd v1-4 as backbone.

**Comment by Yonghao-Yu** [comment]
> @Yonghao-Yu Hello yonghao, I use COCO dataset, the original T2I Adapter with sd v1-4 as backbone.

Yes, the paper mentions using the COCO17 dataset and using pidinet to extract edges. But I don’t know the details, such as the data format, etc. Did you make the dataset yourself? Is there a tutorial I can follow? Or where can I download the dataset so that I can use it directly? Thank you very much!

**Comment by cuhkfz** [comment]
> > @Yonghao-Yu Hello yonghao, I use COCO dataset, the original T2I Adapter with sd v1-4 as backbone.
> 
> Yes, the paper mentions using the COCO17 dataset and using pidinet to extract edges. But I don’t know the details, such as the data format, etc. Did you make the dataset yourself? Is there a tutorial I can follow? Or where can I download the dataset so that I can use it directly? Thank you very much!

Just download the required coco dataset, and run train_sketch.py. The only pre-process I did was create train_color/val_color.

**Comment by Yonghao-Yu** [comment]
> > > @Yonghao-Yu Hello yonghao, I use COCO dataset, the original T2I Adapter with sd v1-4 as backbone.
> > 
> > 
> > Yes, the paper mentions using the COCO17 dataset and using pidinet to extract edges. But I don’t know the details, such as the data format, etc. Did you make the dataset yourself? Is there a tutorial I can follow? Or where can I download the dataset so that I can use it directly? Thank you very much!
> 
> Just download the required coco dataset, and run train_sketch.py. The only pre-process I did was creating train_color/val_color. But the training does not converge.

I see, I'll try again, thanks again.

**Comment by Windrain7** [comment]
> Just download the required coco dataset, and run train_sketch.py. The only pre-process I did was create train_color/val_color.

Hello, could you tell me how to set the parameters of train_sketch.py?

**Comment by Junxix** [comment]
> > > @Yonghao-Yu Hello yonghao, I use COCO dataset, the original T2I Adapter with sd v1-4 as backbone.
> > 
> > 
> > Yes, the paper mentions using the COCO17 dataset and using pidinet to extract edges. But I don’t know the details, such as the data format, etc. Did you make the dataset yourself? Is there a tutorial I can follow? Or where can I download the dataset so that I can use it directly? Thank you very much!
> 
> Just download the required coco dataset, and run train_sketch.py. The only pre-process I did was create train_color/val_color.

Hello, how can I set the configuration since dataset_laion.py only describes the laion's format?

**Comment by ydniuyongjie** [comment]
> Hello,Thank you very much for your open-source work. I have a question regarding the visualization of the loss generated while running train_sketch.py. Below is the visualization of the validation loss. The training reached 290,000 steps and epoch 7, but the loss does not show signs of convergence. I am unsure when the training should be considered complete. Could you please provide some guidance on this? ![loss](https://github.com/user-attachments/assets/60e4023e-d04f-40eb-ba14-42a5d0455d9b)
What is your batch size? You've iterated 290,000 steps, and the epoch is already at 7. How many GPUs are you using?
